### PR TITLE
Touchups after dm deletion bugbash

### PR DESCRIPTION
**Pull request by clcaelynn-clever**
# Jira
[M5G-786: focus-border on Delete button should be properly sized](https://clever.atlassian.net/browse/M5G-786)

# Overview

One commit to improve testability of DM Deletion within Dewey by creating a confirm-delete modal (highly similar to that in launchpad) and using it in the MessagingBubbleView component.

Another commit to resolve an issue with the focus-border size by tweaking the CSS for NormalMessagingBubble.

# Screenshots/

### Chrome
Screen Shot 2021-09-08 at 4 30 01 PM

### Safari
Screen Shot 2021-09-08 at 4 29 49 PM

### Chrome
Screen Shot 2021-09-08 at 4 30 17 PM

### Safari
Screen Shot 2021-09-08 at 4 29 40 PM

# Testing:

- [ ] Unit tests
- Manual tests:
  - [x] Chrome
  - [x] Safari
  - [ ] IE11

# Roll Out:

- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [ ] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT URL LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: src/MessagingBubble/NormalMessagingBubble.less [hunk 1]
```diff
@@ -114,15 +114,14 @@
...
-  max-width: 100%;
 
   &:hover {
     .NormalMessagingBubble--Message--Timestamp {
       display: none;
     }
 
     .NormalMessagingBubble--Message--Delete {
-      display: unset;
+      display: block;
```
Comment (clcaelynn-clever) [inline src/MessagingBubble/NormalMessagingBubble.less:124]: Safer bet than `unset`.

File: src/MessagingBubble/NormalMessagingBubble.less [hunk 2]
```diff
@@ -79,13 +79,13 @@
 }
 
 .NormalMessagingBubble--Message--Metadata--Own {
-  min-width: 3.5rem;
+  min-width: 3.25rem;
```
Comment (clcaelynn-clever) [inline src/MessagingBubble/NormalMessagingBubble.less:82]: `3.5rem` was unnecessary, as far as I could tell.

**Review by cofauver — APPROVED** [review]
Looks good to me!

File: docs/components/MessagingBubbleView.jsx [hunk 3]
```diff
@@ -393,7 +403,7 @@ export default class MessagingBubbleView extends React.PureComponent {
               name: "replyTo",
               type: "React.ReactNode",
               description: "Optional reply-to announcement, if this DM was an announcement reply.",
-              option: true,
+              optional: true,
```
Comment (clcaelynn-clever) [inline docs/components/MessagingBubbleView.jsx:406]: Typo from before.

File: src/MessagingBubble/NormalMessagingBubble.tsx [hunk 4]
```diff
@@ -178,20 +185,29 @@ function _renderNonInteractiveTimestamp({
...
   metadataClassNames: string;
   actionButtonClassNames: string;
 }): React.ReactNode {
+  const formattedTimestamp = _formatDateForTimestamp(timestamp);
+
   return (
     <div className={metadataClassNames}>
-      <Button className={actionButtonClassNames} type="linkPlain" onClick={onClickDeleteButton}>
+      <Button
+        ariaLabel={`Delete ${formattedTimestamp} ${messageBody || ""}`}
```
Comment (clcaelynn-clever) [inline src/MessagingBubble/NormalMessagingBubble.tsx:203]: The `|| ""` avoids having the screen-reader read out "undefined" if the prop isn't in for whatever reason.

File: docs/components/MessagingBubbleView.jsx (hunk 3 above)
Comment (cofauver) [reply docs/components/MessagingBubbleView.jsx:406]: wow, good catch, sorry I missed it in my review!

**Review by cofauver — APPROVED** [review]
Nice catches!

**Comment by clcaelynn-clever** [comment]
(Latest push is just a rebase that reorders commits and fixes to minor version bump instead of patch version bump, to account for new functionality.)

**Pull request merged**